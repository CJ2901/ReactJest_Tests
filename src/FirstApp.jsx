import PropTypes from 'prop-types';

export const FirstApp = ( { title , subTitle, name } ) => {

    return (
        <>
            <h1 data-testid='test-title'> Hola, soy { title } </h1>
            {/* <code>{JSON.stringify(objetoMensaje)}</code> */}
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p>{ name }. Derechos reservados</p>
        </>
    )

}

FirstApp.propTypes = {
    subTitle: PropTypes.string,
    title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    name: 'Alexander Magallanes',
    subTitle: 'No hay subt√≠tulo en este episodio',
    title: 'el PoderMaximo',
}
