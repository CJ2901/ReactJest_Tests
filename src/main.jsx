import React from 'react';
import ReactDOM from 'react-dom/client'
import { Counterapp } from './Counterapp';
import { FirstApp } from './FirstApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title="The Batman" subTitle="The Dark Knight" /> */}
        <Counterapp value={10}/>
    </React.StrictMode>

);