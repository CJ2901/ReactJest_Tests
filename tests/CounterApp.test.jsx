import { fireEvent, render, screen } from "@testing-library/react";
import { Counterapp } from "../src/Counterapp"


describe('Pruebas en CounterApp.test.jsx ', () => { 

    const initialValue = 100;
    
    test('should hacer match con el snapshot', () => { 
        
        const {container} = render( <Counterapp value={initialValue} /> )
        expect( container ).toMatchSnapshot();

     });

    test('should mostrar el valor inicial de 100', () => { 
        
        render( <Counterapp value={initialValue} /> );
        expect( screen.getByText(initialValue ) ).toBeTruthy();

     });
     
     test('should incrementar con el botón +1', () => { 
         
         render( <Counterapp value={initialValue} /> );
        //  fireEvent.click( screen.getByText('+1') );
         fireEvent.click( screen.getByRole('button', {name: 'btn-more'}));
         expect( screen.getByText(101) ).toBeTruthy();
 
      });
      
      test('should decrementar con el botón -1', () => { 
          
          render( <Counterapp value={initialValue} /> );
        //   fireEvent.click( screen.getByText('-1') );
          fireEvent.click( screen.getByRole('button', {name: 'btn-less'}));
          expect( screen.getByText(99) ).toBeTruthy();
          
       });
      
      test('should funcionar el botón de reset', () => { 
          
          render( <Counterapp value={initialValue} /> );
          fireEvent.click( screen.getByText('+1') );
          fireEvent.click( screen.getByText('+1') );
          fireEvent.click( screen.getByText('+1') );

          fireEvent.click( screen.getByRole('button', {name: 'btn-reset'}));


          expect( screen.getByText(initialValue) ).toBeTruthy();

       });

 });