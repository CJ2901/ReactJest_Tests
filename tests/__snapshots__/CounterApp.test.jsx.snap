// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp.test.jsx  should hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    aria-label="btn-more"
  >
     +1 
  </button>
  <button
    aria-label="btn-less"
  >
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
